src/utils/validators/addWallet.ts

```diff
@@ -58,6 +58,32 @@ export const isNameUnique = server$(async function (name: string) {
   return true;
 });
 
+export const UniqueAddressResult = z.array(z.array(z.string()));
+export const isAddressUnique = server$(async function (address: string) {
+  const db = await connectToDB(this.env);
+  const cookie = this.cookie.get("accessToken");
+  if (!cookie) {
+    throw new Error("No cookie found");
+  }
+  const { userId } = jwt.decode(cookie.value) as JwtPayload;
+  const result = (
+    await db.query(`
+      LET $walletId = SELECT VALUE ->observes_wallet.out FROM ${userId};
+      SELECT VALUE address FROM $walletId;
+    `)
+  ).at(1);
+  if (!result) {
+    return true;
+  }
+  const [usersObservedWallets] = UniqueAddressResult.parse(result);
+  for (const observedWalletAddress of usersObservedWallets) {
+    if (observedWalletAddress === address) {
+      return false;
+    }
+  }
+  return true;
+});
+
```

Comment:
I believe we could simplify this query to:
`SELECT count() FROM wallet WHERE address = type::string(address)"
This should work, lmk if im wrong

Reply:
updated - made it slightly different though. Your query is not user scoped.